sklearn/compose/_column_transformer.py

```diff
@@ -332,19 +336,22 @@ def get_feature_names(self):
         """
         check_is_fitted(self, 'transformers_')
         feature_names = []
-        for name, trans, _, _ in self._iter(fitted=True):
+        for dim, (name, trans, column, _) in \
+                zip(self._output_dims, self._iter(fitted=True)):
             if trans == 'drop':
-                continue
+                pass
             elif trans == 'passthrough':
-                raise NotImplementedError(
-                    "get_feature_names is not yet supported when using "
-                    "a 'passthrough' transformer.")
+                if name == 'remainder':
+                    feature_names.extend(self._remainder_names)
+                else:
+                    feature_names.extend([name + "__" + str(c)
+                                          for c in column])
             elif not hasattr(trans, 'get_feature_names'):
-                raise AttributeError("Transformer %s (type %s) does not "
-                                     "provide get_feature_names."
-                                     % (str(name), type(trans).__name__))
-            feature_names.extend([name + "__" + f for f in
-                                  trans.get_feature_names()])
+                feature_names.extend([name + "__x" + str(i)
+                                      for i in range(dim)])
```

Comment:
So this is also giving names if the transformer has no feature names method? (just to clarify, this is not strictly related to the PR of enabling feature names for passthrough?)

Reply:
Yes that's right, it seems better to give default names than to error out especially if several of the other transformers have feature names. And even if none of them do, at least this let's you know which features came from which transformer. I suppose it isn't strictly related to the passthrough parameter, although is pretty similar. Would it be worth making a separate pull request for this?

Reply:
Hmm... Yes, I don't think we should be doing this. I'd rather tell the user that they have to define get_feature_names in a component transformer than to invent something.

Reply:
Fair enough, I suppose this would have just been a workaround until every Transformer has a get_feature_names() method.

Reply:
I've now removed this change